### PR TITLE
Add small doc section about building in docker container

### PR DESCRIPTION
**Pull request by neumantm**
Resolves: #1088

### CODE REVIEW

File: docs/source/user/build_spack.rst [hunk 1]
```diff
@@ -211,3 +211,21 @@ the dependencies by using ``spack location -i <package>``. For example in the
...
+
+::
+
+   docker run -it ecpe4s/exawind-snapshot bash
+
+Inside the container you can directly install Nalu-Wind with spack:
+
+::
+
+   spack install nalu-wind
```
Comment (psakievich) [inline docs/source/user/build_spack.rst:231]:
```suggestion
   spack install nalu-wind
   
 `Spack-Manager <https://github.com/sandialabs/spack-manager>`_ is used generate the container and is pre-installed in the container.  Additional development of the code can be done through the Spack-Manager's developer tools.  A quick-start guide for these tools is available `here <https://sandialabs.github.io/spack-manager/user_profiles/developers/developer_workflow.html>`_.  This workflow is how the CI builds are generated in GitHub actions.
 
If you are uncomfortable using spack for development then you can also set up your own build workflow in more of a "roll-your-own" development environment using CMake and just utilize the pre-installed TPL's that are available in the container through spack.  This really comes down to your development preferences and your comfort level with CMake.
```

Technically `nalu-wind` is already installed in the container so doing `spack install nalu-wind` should just re-use that install.  I've added some documentation on how you can use `spack-manager` to do some extra development.

**Review by psakievich — CHANGES_REQUESTED** [review]
Thanks for getting this started.  It's super helpful when people add documentation right after they figure things out.  I've added some suggested text. We can iterate on them further as needed.

File: docs/source/user/build_spack.rst (hunk 1 above)
Comment (neumantm) [reply docs/source/user/build_spack.rst:231]: Ah right, didn't notice that. Then maybe we should write `spack load nalu-wind` (indicating it is already installed) instead?

**Review by psakievich — APPROVED** [review]
LGTM. `spack load nalu-wind` seems like a nice addition that would be more semantically correct

File: docs/source/user/build_spack.rst [hunk 2]
```diff
@@ -211,3 +211,25 @@ the dependencies by using ``spack location -i <package>``. For example in the
...
+It is also possible to build (and run) Nalu-Wind inside a docker container with a prepared environment.
+This has the benefit of requiring less setup and usually being faster and can be useful in some situations (e.g. quickly testing something).
+
+For this, use the container `ecpe4s/exawind-snapshot <https://hub.docker.com/r/ecpe4s/exawind-snapshot>`_:
+
+::
+
+   docker run -it ecpe4s/exawind-snapshot bash
+
+Inside the container you can directly install Nalu-Wind with spack:
```
Comment (psakievich) [inline docs/source/user/build_spack.rst:227]:
```suggestion
Inside the container you can directly load the pre-installed version of Nalu-Wind with spack:
```

**Pull request merged**